src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._form = this._popupElement.querySelector(".modal__form");
+    this._deleteSubmitBtn = this._popupElement.querySelector("#confirm-button");
+  }
+
+  setSubmitAction(handler) {

```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.